lib/gst/player/gstplayer.c

```diff
@@ -656,6 +658,21 @@ error_cb (GstBus * bus, GstMessage * msg, gpointer user_data)
   g_mutex_unlock (&self->priv->lock);
 }
 
+static void
+element_cb (GstBus * bus, GstMessage * msg, gpointer user_data)
+{
+  GstPlayer *self = GST_PLAYER (user_data);
+
+  if (gst_is_missing_plugin_message (msg)) {
+    gchar *desc;
+
+    desc = gst_missing_plugin_message_get_description (msg);
+    emit_error (self, g_error_new (GST_PLAYER_ERROR,
+        GST_PLAYER_ERROR_MISSING_PLUGIN, "Missing plugin '%s'", desc));
+    g_free (desc);
+  }
```

Comment:
This is almost ok, but note that there can be multiple missing-plugin messages in theory. And only after the last one an error message is posted, or a warning message if only irrelevant plugins are missing and playback can continue.

I think you should accumulate all missing plugin messages and wait until the error or warning message (GST_CORE_ERROR_MISSING_PLUGIN or GST_STREAM_ERROR_CODEC_NOT_FOUND). And only if it's an error message, actually emit an error. As emitting an error will stop playback, which might not be needed.
In case of a warning message, it might make sense to add a warning signal that mirrors the error signal but has the big difference that it does not cause playback to stop.